guides/material-2.md

```diff
@@ -0,0 +1,657 @@
+# Theming Angular Material 2
+
+## What is theming?
+
+Angular Material's theming system lets you customize base, color, typography, and density styles for
+components in your application. The theming system is based on Google's
+[Material Design 2][material-design-theming] specification.
+
+**This guide refers to Material 2, the previous version of Material. For the
+latest Material 3 documentation, see the [theming guide][theming].**
+
+**For information on how to update, see the section
+on [how to migrate an app from Material 2 to Material 3](#how-to-migrate-an-app-from-material-2-to-material-3).**
+
+This document describes the concepts and APIs for customizing colors. For typography customization,
+see [Angular Material Typography][mat-typography]. For guidance on building components to be
+customizable with this system, see [Theming your own components][theme-your-own].
+
+[material-design-theming]: https://m2.material.io/design/material-theming/overview.html
+[theming]: https://material.angular.io/guide/theming
+[mat-typography]: https://material.angular.io/guide/typography
+[theme-your-own]: https://material.angular.io/guide/theming-your-components
+
+### Sass
+
+Angular Material's theming APIs are built with [Sass](https://sass-lang.com). This document assumes
+familiarity with CSS and Sass basics, including variables, functions, and mixins.
+
+You can use Angular Material without Sass by using a pre-built theme, described in
+[Using a pre-built theme](#using-a-pre-built-theme) below. However, using the library's Sass API
+directly gives you the most control over the styles in your application.
+
+## Palettes
+
+A **palette** is a collection of colors representing a portion of color space. Each value in this
+collection is called a **hue**. In Material Design, each hues in a palette has an identifier number.
+These identifier numbers include 50, and then each 100 value between 100 and 900. The numbers order
+hues within a palette from lightest to darkest.
+
+Angular Material represents a palette as a [Sass map][sass-maps]. This map contains the
+palette's hues and another nested map of contrast colors for each of the hues. The contrast colors
+serve as text color when using a hue as a background color. The example below demonstrates the
+structure of a palette. [See the Material Design color system for more background.][spec-colors]
+
+```scss
+$indigo-palette: (
```

Comment:
In #28892 all these APIs were prefixed with `m2-`. We should update the guide to reflect that.

Reply:
Fixed just in the M2 guide, will keep this in mind when changing the other guides in a separate PR, thanks!